### PR TITLE
Enable/Disable creation right for Workgroup or Drive

### PR DESCRIPTION
**Pull request by hoangdat**
@dab246 Please review 

https://user-images.githubusercontent.com/6462404/142968954-ee558667-6172-46f2-bc82-d15e5356012a.mov

### CODE REVIEW

File: lib/presentation/widget/shared_space/shared_space_widget.dart [hunk 1]
```diff
@@ -203,6 +196,25 @@ class _SharedSpaceWidgetState extends State<SharedSpaceWidget> {
...
+  }
+
+  void _onPressCreateNewButton(BuildContext context, AppState appState) {
+    if (appState.functionalityState.isDriveEnable() && appState.functionalityState.isDriveCreationEnable() &&
+      appState.functionalityState.isCreateWorkgroupEnable()) {
+      sharedSpaceViewModel.openCreateNewSharedSpaceMenu(context, _createNewSharedSpaceActionTiles(context));
+    } else if (appState.functionalityState.isDriveEnable() && appState.functionalityState.isDriveCreationEnable() &&
+      !appState.functionalityState.isCreateWorkgroupEnable()) {
+      sharedSpaceViewModel.openCreateNewDrive(context);
+    } else if (!appState.functionalityState.isDriveCreationEnable() && appState.functionalityState.isCreateWorkgroupEnable()) {
```
Comment (dab246) [inline lib/presentation/widget/shared_space/shared_space_widget.dart:213]: IMO U are missing the case where Workgroup is disabled

**Review by dab246 — APPROVED** [review]

File: lib/presentation/widget/shared_space/shared_space_widget.dart (hunk 1 above)
Comment (hoangdat) [reply lib/presentation/widget/shared_space/shared_space_widget.dart:213]: 1. drive and workgroup are enabled
2. drive enable and workgroup disable
3. drive disable and workgroup enable
4. both are disable -> no action

Comment (dab246) [reply lib/presentation/widget/shared_space/shared_space_widget.dart:213]: You just checked `isCreateWorkgroupEnable ` enabled. I don't see a check for `isSharedSpaceEnable `

Comment (hoangdat) [reply lib/presentation/widget/shared_space/shared_space_widget.dart:213]: ah, TBH, the creation right is not related to the Workgroup enable or not, it is the same for Drive :)

Comment (dab246) [reply lib/presentation/widget/shared_space/shared_space_widget.dart:213]: If so you just need to check `isDriveCreationEnable` without `isDriveEnable` in this case.

Comment (hoangdat) [reply lib/presentation/widget/shared_space/shared_space_widget.dart]: please check again

Comment (dab246) [reply lib/presentation/widget/shared_space/shared_space_widget.dart]: Done

**Pull request merged**